pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -144,21 +144,31 @@ def capture_compile(*, host_install_dir: str, skip_host: bool, env: dict[str, st
     env["PATH"] = str(TOOLSDIR) + ":" + env["PATH"]
     capture_make_command_wrapper_symlinks(env)
 
-    cmd = [sys.executable, "setup.py"]
     if skip_host:
         env["SKIP_HOST"] = "1"
-        cmd.append("build")
-    else:
-        assert host_install_dir, "Missing host_install_dir"
-        cmd.extend(["install", "--home", host_install_dir])
 
-    result = subprocess.run(cmd, env=env)
-    if result.returncode != 0:
+    try:
+        subprocess.check_call([sys.executable, "setup.py", "bdist_wheel"], env=env)
+        if not skip_host:
+            assert host_install_dir, "Missing host_install_dir"
+            result_wheel = str(list(Path("dist").glob("*.whl"))[0])
+            subprocess.check_call(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
```

Comment:
The only thing is pip will often use build isolation to install build requirements from `pyproject.toml` and not whatever dependencies we indicated in our build system.
In most cases, that's probably fine, but I wonder if this won't be an issue if the build dependency is patched. Say you build a package that needs numpy at build time, and it will use a different (and unpatched) numpy version to install it on the host when skip_host=False. 
Again I'm not sure if it's really an issue, but we could at least mention it in a comment that in some cases it might be appropriate to disable build isolation here.

Reply:
Oh yeah, this was also removed in #2256 (along with the entire `skip_host` key). Only `numpy` was setting `skip_host` to false too, so if there were problems on this code path we probably wouldn't know.